components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1183,8 +1183,11 @@ where
             SignificantMsg::RaftLogGcFlushed => {
                 self.on_raft_log_gc_flushed();
             }
-            SignificantMsg::RaftlogFetched { context, .. } => {
+            SignificantMsg::RaftlogFetched { context, res } => {
+                self.fsm.peer.mut_store().set_async_fetch_res(Some(res));
                 self.fsm.peer.raft_group.on_entries_fetched(context);
+                // clean the async fetch result immediately if not used to free memory
+                self.fsm.peer.mut_store().set_async_fetch_res(None);
```

Comment:
`async_fetch_res` is set and clean immediately, is it possible that other places can still see `async_fetch_res = RaftlogFetchResult::Fetched`?